// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: pb/graphql.proto

package graphql

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on Field with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *Field) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Field with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in FieldMultiError, or nil if none found.
func (m *Field) ValidateAll() error {
	return m.validate(true)
}

func (m *Field) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Required

	// no validation rules for Params

	// no validation rules for Dirs

	// no validation rules for Ignore

	// no validation rules for Name

	if len(errors) > 0 {
		return FieldMultiError(errors)
	}

	return nil
}

// FieldMultiError is an error wrapping multiple validation errors returned by
// Field.ValidateAll() if the designated constraints aren't met.
type FieldMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m FieldMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m FieldMultiError) AllErrors() []error { return m }

// FieldValidationError is the validation error returned by Field.Validate if
// the designated constraints aren't met.
type FieldValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e FieldValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e FieldValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e FieldValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e FieldValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e FieldValidationError) ErrorName() string { return "FieldValidationError" }

// Error satisfies the builtin error interface
func (e FieldValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sField.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = FieldValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = FieldValidationError{}

// Validate checks the field values on Rpc with the rules defined in the proto
// definition for this message. If any rules are violated, the first error
// encountered is returned, or nil if there are no violations.
func (m *Rpc) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Rpc with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in RpcMultiError, or nil if none found.
func (m *Rpc) ValidateAll() error {
	return m.validate(true)
}

func (m *Rpc) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Type

	// no validation rules for Ignore

	// no validation rules for Name

	if len(errors) > 0 {
		return RpcMultiError(errors)
	}

	return nil
}

// RpcMultiError is an error wrapping multiple validation errors returned by
// Rpc.ValidateAll() if the designated constraints aren't met.
type RpcMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m RpcMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m RpcMultiError) AllErrors() []error { return m }

// RpcValidationError is the validation error returned by Rpc.Validate if the
// designated constraints aren't met.
type RpcValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e RpcValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e RpcValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e RpcValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e RpcValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e RpcValidationError) ErrorName() string { return "RpcValidationError" }

// Error satisfies the builtin error interface
func (e RpcValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sRpc.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = RpcValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = RpcValidationError{}

// Validate checks the field values on Svc with the rules defined in the proto
// definition for this message. If any rules are violated, the first error
// encountered is returned, or nil if there are no violations.
func (m *Svc) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Svc with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in SvcMultiError, or nil if none found.
func (m *Svc) ValidateAll() error {
	return m.validate(true)
}

func (m *Svc) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Type

	// no validation rules for Ignore

	// no validation rules for Name

	if len(errors) > 0 {
		return SvcMultiError(errors)
	}

	return nil
}

// SvcMultiError is an error wrapping multiple validation errors returned by
// Svc.ValidateAll() if the designated constraints aren't met.
type SvcMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SvcMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SvcMultiError) AllErrors() []error { return m }

// SvcValidationError is the validation error returned by Svc.Validate if the
// designated constraints aren't met.
type SvcValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SvcValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SvcValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SvcValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SvcValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SvcValidationError) ErrorName() string { return "SvcValidationError" }

// Error satisfies the builtin error interface
func (e SvcValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSvc.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SvcValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SvcValidationError{}
